// obs-websocket-js/dist/msgpack.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); var _class;





var _chunkXS3U5T4Qcjs = require('./chunk-XS3U5T4Q.cjs');

// src/msgpack.ts
var _msgpack = require('@msgpack/msgpack');
var OBSWebSocket = (_class = class extends _chunkXS3U5T4Qcjs.BaseOBSWebSocket {constructor(...args) { super(...args); _class.prototype.__init.call(this); }
  __init() {this.protocol = "obswebsocket.msgpack"}
  async encodeMessage(data) {
    return _msgpack.encode.call(void 0, data);
  }
  async decodeMessage(data) {
    if (typeof Blob !== "undefined" && data instanceof Blob) {
      data = await data.arrayBuffer();
    }
    return _msgpack.decode.call(void 0, data);
  }
}, _class);
var msgpack_default = OBSWebSocket;







exports.EventSubscription = _chunkXS3U5T4Qcjs.EventSubscription; exports.OBSWebSocket = OBSWebSocket; exports.OBSWebSocketError = _chunkXS3U5T4Qcjs.OBSWebSocketError; exports.RequestBatchExecutionType = _chunkXS3U5T4Qcjs.RequestBatchExecutionType; exports.WebSocketOpCode = _chunkXS3U5T4Qcjs.WebSocketOpCode; exports.default = msgpack_default;
